JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -86,22 +85,16 @@ public SqsNotificationListener WithMessageProcessingStrategy(IMessageProcessingS
             _handlerMap.Add(typeof(T), handlerFunc);
         }
 
-        public void Listen()
+        public void Listen(CancellationToken cancellationToken = default)
         {
             var queue = _queue.QueueName;
             var region = _queue.Region.SystemName;
             var queueInfo = $"Queue: {queue}, Region: {region}";
 
-            _cts = new CancellationTokenSource();
-            Task.Factory.StartNew(async () =>
-                {
-                    while (!_cts.IsCancellationRequested)
-                    {
-                        await ListenLoop(_cts.Token).ConfigureAwait(false);
-                    }
-                })
+            Task.Factory.StartNew(async () => { await ListenLoop(cancellationToken).ConfigureAwait(false); },
+                    cancellationToken)
```

Comment:
Could you remove this cancellation token, again, this bit of code can be addressed in a separate PR

Reply:
Done